applications/Anchor.md

```diff
@@ -0,0 +1,286 @@
+
+- There are 4 methods and 2 stores to implement the on-chain linked list function in Anchor pallet.
+
+- Substrate 3.0.0 and Rust for Anchor pallet. Javascript for anchor.js.
+
+- Code here [https://github.com/ff13dfly/Anchor](https://github.com/ff13dfly/Anchor), the document is the next step.
+
+- Treat Anchor as Name Service, there is no available pallet can be use. Have checked here [Open Source Polkadot Stack](https://wiki.polkadot.network/docs/build-open-source).
+![easypolka.png](http://android.im/anchor/status.png)
+
+- The function of Anchor has been finalized and will not be further expanded on substrate side. The function of the current version is the final form.
+
+### Ecosystem Fit
+
+- An available Name Services on Substrate 3.0.0, and it is extended to a On-chain Linked List system.
+
+- Developers who have not yet used substrate/Polkadot. Developers who do not like Smart Contract way to develop application.
+
+- Developer can build application flexibly without understanding the whole blockchain system. I think this is attractive to many developers.
+
+- From the [Open Source Polkadot Stack](https://wiki.polkadot.network/docs/build-open-source), there are 4 Name Service pallets, 2 red ( can not compatible to substrate 3.0.0 ) and 2 yellow ( code not updated more than 2 months by checking github ).
+
+## Team :busts_in_silhouette:
+
+### Team members
+
+- Zhongqiang Fu, individual developer.
+
+### Contact
+
+- **Contact Name:** Zhongqiang Fu
+- **Contact Email:** ff13dfly@163.com
+- **Website:** https://github.com/ff13dfly/
+
+### Legal Structure
+
+- Individual
+
+### Team's experience
+
+- On substrate, Substrate with Anchor pallet has been build successful and run at [wss:dev.metanchor.net](wss:dev.metanchor.net). I have tried to load a three nodes network successful.
+
+### Team Code Repos
+
+- https://github.com/ff13dfly/
+- https://github.com/ff13dfly/Anchor
+
+### Team LinkedIn Profiles (if available)
+
+## Development Status :open_book:
+
+- Demo cApp [freeSaying](https://android.im/vManager/) is published now. The test network is available, you can access [wss:dev.metanchor.net](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fdev.metanchor.net#/explorer). Anchor pallet functions have been developed, but not tested entirely.
+
+## Development Roadmap :nut_and_bolt:
+
+### Anchor pallet
+
+- There are 4 methods exposed. One for setting data, Three for trading anchor.
+  ![easypolka.png](http://android.im/anchor/methods.png)
+  1. setAnchor, set Anchor data method.
+
+    ```RUST
+    pub fn set_anchor(
+      origin: OriginFor<T>,   //default
+      key: Vec<u8>,           //Anchor name
+      raw: Vec<u8>,           //raw data to storage
+      protocol: Vec<u8>,      //data protocol, used to decide how to decode raw data
```

Comment:
What is `protocol` here and what does it do? Can you give an example?

Reply:
> What is `protocol` here and what does it do? Can you give an example?

There are two samples that I used for the cApp freeSaying.net.

1. cApp protocol.
```Javascript
{
  "key":"any_name",
  "raw":"JS string above",
  "protocol": JSON.stringify({"type":"app","format":"JS","ver":"0.0.1"}),
  "pre":0
}
```

2. data protocol
```Javascript
{
  "key":"any_name",
  "raw":JSON.stringify({"any":"object"}),
  "protocol": JSON.stringify({"type":"data","format":"JSON"}),
  "pre":0
}
```

cApp can subscribe the latest block and decode the data , if the protocol is supported by the cApp, it can be used by the cApp.
In short, the protocol is highly customized for the developer.